### ISSUE TITLE
Filebrowser only works for directory depth = 1

### ISSUE THREAD

**Issue by bartkamphorst** [repotag/repotag#18]
For example, `<reponame>/app/config/locales` is the deepest the file browser will go, and at that point it will show the `<reponame>/app/config` directory.

**Comment by bartkamphorst** [comment]
It seems that `@directory_list` and `@file_list` are not updated properly. It has lead me to [this helper method](https://github.com/repotag/repotag/blob/master/app/helpers/repositories_helper.rb#L28) but it might actually lead into the `RJGit::Porcelain.ls_tree` implementation.

**Comment by dometto** [comment]
It looks like that helper is simply not setting the `ls_tree` method's `options[:file_path]` option, so it will always default to the root directory. (Another strategy would be to `ls_tree` recursively to some level and cache the results, so we don't need to do a new call everytime the user clicks a directory.)

By the way, the helper is setting the `:branch` option, but I think this should be `:ref` according to [this](https://github.com/repotag/rjgit/blob/master/lib/rjgit.rb#L69).

**Comment by bartkamphorst** [comment]
> It looks like that helper is simply not setting the ls_tree method's options[:file_path]

Would we need to, given that a tree is passed in with a relative path?

>  think this should be :ref

Good catch. Changed this locally.

**Comment by dometto** [comment]
Good point about the tree. I guess it's going wrong in constructing the tree from @current_path in repositories_controller.rb:57 then. Do we need the extra stap of constructing the tree though, if we can theoretically just pass a :file_path to #lstree?
